server/routes/api/user.ts

```diff
@@ -34,242 +36,300 @@ let postApplicationBranchErrorHandler: express.ErrorRequestHandler = (err, reque
-	let requestType: ApplicationType = request.url.match(/\/application\//) ? ApplicationType.Application : ApplicationType.Confirmation;
-	let branchName = request.params.branch as string;
-	let branch = (await Branches.BranchConfig.loadAllBranches()).find(b => b.name.toLowerCase() === branchName.toLowerCase()) as (Branches.ApplicationBranch | Branches.ConfirmationBranch);
-	if (!branch) {
-		response.status(400).json({
+// TODO what is the difference between this and `middleware.timeLimited`? - related to #206
+function applicationTimeRestriction(requestType: ApplicationType): express.RequestHandler {
+	return async (request, response, next) => {
+		let branchName = request.params.branch as string;
+		let branch = (await Branches.BranchConfig.loadAllBranches()).find(b => b.name.toLowerCase() === branchName.toLowerCase()) as (Branches.ApplicationBranch | Branches.ConfirmationBranch);
+		if (!branch) {
+			response.status(400).json({
 			"error": "Invalid application branch"
-		});
-		return;
-	}
-
-	let user = request.user as IUserMongoose;
-
-	let openDate = branch.open;
-	let closeDate = branch.close;
-	if (requestType === ApplicationType.Confirmation && user.confirmationDeadlines) {
-		let times = user.confirmationDeadlines.find((d) => d.name === branch.name);
-		if (times) {
-			openDate = times.open;
-			closeDate = times.close;
+			});
+			return;
 		}
-	}
 
-	if (moment().isBetween(openDate, closeDate) || request.user.isAdmin) {
-		next();
-	}
-	else {
-		response.status(408).json({
-			"error": `${requestType === ApplicationType.Application ? "Applications" : "Confirmations"} are currently closed`
-		});
-		return;
-	}
-};
+		let user = request.user as IUserMongoose;
 
-userRoutes.route("/application/:branch")
-	.post(isUserOrAdmin, applicationTimeRestriction, postParser, uploadHandler.any(), postApplicationBranchErrorHandler, postApplicationBranchHandler)
-	.delete(isUserOrAdmin, applicationTimeRestriction, deleteApplicationBranchHandler);
-userRoutes.route("/confirmation/:branch")
-	.post(isUserOrAdmin, applicationTimeRestriction, postParser, uploadHandler.any(), postApplicationBranchErrorHandler, postApplicationBranchHandler)
-	.delete(isUserOrAdmin, applicationTimeRestriction, deleteApplicationBranchHandler);
+		let openDate = branch.open;
+		let closeDate = branch.close;
+		if (requestType === ApplicationType.Confirmation && user.confirmationDeadlines) {
+			let times = user.confirmationDeadlines.find((d) => d.name === branch.name);
+			if (times) {
+				openDate = times.open;
+				closeDate = times.close;
+			}
+		}
 
-async function postApplicationBranchHandler(request: express.Request, response: express.Response): Promise<void> {
-	let requestType: ApplicationType = request.url.match(/\/application\//) ? ApplicationType.Application : ApplicationType.Confirmation;
+		if (moment().isBetween(openDate, closeDate) || request.user.isAdmin) {
+			next();
+		}
+		else {
+			response.status(408).json({
```

Comment:
Is a request timeout the appropriate status code for this?

Reply:
you are a sassyboi

Reply:
literally you did this